add project description (and update pre-commit config)

tanikina opened the pull request:
This PR adds a brief project description, an illustration of the "probing" workflow and a citation for the current arXiv publication that needs to be updated after the proceedings are out.

I also updated our pre-commit config because the previous version seems to be broken due to the docstring formatting, originally we had docstring v1.4 in `.pre-commit-config.yaml` which results in the following error message:
```
    pre_commit.clientlib.InvalidManifestError: 
=====> /home/username/.cache/pre-commit/repo_d3cgo1i/.pre-commit-hooks.yaml is not a file
```
I also had to specify an older version of wandb-0.16.0 in `requirements.txt` that allows to import `wandb.wandb_torch`, the current one (wandb-0.19.8) does not support this, unfortunately, and the tests fail.

Another issue was caused by `AttributeError: np.float_ was removed in the NumPy 2.0 release. Use np.float64 instead.` that appeared when running the tests. So, for now I simply added the older version of numpy to `requirements.txt`, the code works with `numpy==1.24.1`. Or should I rather update everything to make it work with NumPy 2.0?

EDIT: Do we need to remove DFKI-related info from the setup description (i.e. where to find the data on the cluster etc.)?

Important! This repository needs to be made public, so that we can add a working link to it in the paper :)

PS: any improvement suggestions, feedback and comments are very welcome!

tanikina committed:
update pre-commit config

tanikina committed:
update formatting

tanikina committed:
add image

tanikina committed:
update README (add project description)

tanikina committed:
add citation

tanikina committed:
adjust image size and position

tanikina committed:
correct workshop name

tanikina committed:
use old wandb version to import wandb_torch

tanikina committed:
use numpy 1.24.1

tanikina committed:
fix the publication year

ArneBinder reviewed (approved):
> I also updated our pre-commit config because the previous version seems to be broken due to the docstring formatting,

👍

> I also had to specify an older version of wandb-0.16.0 in requirements.txt that allows to import wandb.wandb_torch, the current one (wandb-0.19.8) does not support this, unfortunately, and the tests fail.

👍

> Another issue was caused by AttributeError: np.float_ was removed in the NumPy 2.0 release. Use np.float64 instead. that appeared when running the tests. So, for now I simply added the older version of numpy to requirements.txt, the code works with numpy==1.24.1. Or should I rather update everything to make it work with NumPy 2.0?

no, using numpy<2.0.0 should be fine, upgrading sounds like pain

> EDIT: Do we need to remove DFKI-related info from the setup description (i.e. where to find the data on the cluster etc.)?

I would not bother

> Important! This repository needs to be made public, so that we can add a working link to it in the paper :)

done

> PS: any improvement suggestions, feedback and comments are very welcome!

1. There is a command in the readme that uses a script `usrun.sh` but I do not see it anywhere.
2. Would it be possible to mention the commands to execute the actual experiments? Considering the current readme it is not really clear how to reproduce the results from the paper (but maybe I overlooked something).

tanikina committed:
add a note about usrun.sh

tanikina committed:
add a note on how to reproduce the results

tanikina commented:
Thanks for the feedback!
     
> 1. There is a command in the readme that uses a script usrun.sh but I do not see it anywhere.

Right, I added a short description in the README file. `usrun.sh` is simply a wrapper that specifies the container-mounts, container-image etc. in a separate script. One can also use a standard `srun` command, of course, I just copied it from my command line and forgot that it was a bit custom, sorry.

> 2. Would it be possible to mention the commands to execute the actual experiments? Considering the current readme it is not really clear how to reproduce the results from the paper (but maybe I overlooked something).

This is a bit tricky since we had a LOT of different experiments that are quite well documents in both [`results/coref.md`](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/results/coref.md) and [log.md](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md) which are linked to each other. I added a brief explanation [here](https://github.com/Cora4NLP/multi-task-knowledge-transfer/tree/project_description#how-to-reproduce-our-results). Let me know if this is sufficient or whether I should provide more precise instructions. I believe that it should be possible to reproduce our results by simply following the links in the tables from [`results/coref.md`](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/results/coref.md). All configurations are already available in [`configs`](https://github.com/Cora4NLP/multi-task-knowledge-transfer/tree/main/configs).

tanikina committed:
correct typos
